### PR TITLE
Port dsg 4.27 changes

### PR DESCRIPTION
**Pull request by omytrofanov**
Port changes from dsg-4_27 to 4.30

### CODE REVIEW

**Comment by ybova** [comment]
This change is missed (that is, unassociatedEditorStrategy extension point needs to be removed).
https://github.com/Halliburton-Landmark/eclipse.platform.ui/commit/7a709b665995799e18c4ecac058cf83bfe2a529e#diff-b689016bd7c8fe7b4431afc0fc3781a5fa3e15058120a14d01163bdf4d72b401L15

**Review by ybova — APPROVED** [review]

File: bundles/org.eclipse.e4.ui.css.swt/src/org/eclipse/e4/ui/internal/css/swt/ICTabRendering.java
Comment (hb91546) [inline bundles/org.eclipse.e4.ui.css.swt/src/org/eclipse/e4/ui/internal/css/swt/ICTabRendering.java:1]: Are the method additions found in https://github.com/Halliburton-Landmark/eclipse.platform.ui/pull/38/files#diff-e3bd915de472985f748e90f43ff292c32d10e21aac794ddfc2993e2ef25e4669 intentionally omitted?

File: bundles/org.eclipse.e4.ui.workbench.addons.swt/src/org/eclipse/e4/ui/workbench/addons/cleanupaddon/CleanupAddon.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 package org.eclipse.e4.ui.workbench.addons.cleanupaddon;
 
+import jakarta.annotation.PostConstruct;
```
Comment (hb91546) [inline bundles/org.eclipse.e4.ui.workbench.addons.swt/src/org/eclipse/e4/ui/workbench/addons/cleanupaddon/CleanupAddon.java:18]: import javax.annotation.PostConstruct;
import javax.inject.Inject;

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/TabMenuAddon.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
+package org.eclipse.ui.internal.e4.compatibility;
+
+import jakarta.inject.Inject;
```
Comment (hb91546) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/TabMenuAddon.java:3]: import javax.inject.Inject;

File: bundles/org.eclipse.e4.ui.workbench.addons.swt/src/org/eclipse/e4/ui/workbench/addons/cleanupaddon/CleanupAddon.java (hunk 1 above)
Comment (ybova) [reply bundles/org.eclipse.e4.ui.workbench.addons.swt/src/org/eclipse/e4/ui/workbench/addons/cleanupaddon/CleanupAddon.java:18]: The Eclipse SDK migrated from the javax namespace to the jakarta.
See https://eclipse.dev/eclipse/news/4.30/platform.php#support-jakarta-annotations

File: bundles/org.eclipse.e4.ui.css.swt/src/org/eclipse/e4/ui/internal/css/swt/ICTabRendering.java
Comment (ybova) [reply bundles/org.eclipse.e4.ui.css.swt/src/org/eclipse/e4/ui/internal/css/swt/ICTabRendering.java:1]: We abandoned that CSS customization while migrating to 4.27 - #40

**Pull request merged**